src/index.js

```diff
@@ -1,79 +1,28 @@
-import Books from './modules/Classes.js';
-import Add from './view/add.js';
-import Contact from './view/contact.js';
-import Show from './view/show.js';
+import Books from './modules/Books.js';
+import formatTs from './modules/DateTime.js';
+import { load, showOnPage } from './modules/HashRouter.js';
 // get books data from localStorage if it is null give empty array to class
 const books = new Books((JSON.parse(localStorage.getItem('books')) === null) ? [] : JSON.parse(localStorage.getItem('books')));
 
-// avaliable routes
-const routes = [
-  { path: '/', View: Add },
-  { path: 'list', View: Show },
-  { path: 'about', View: Contact },
-  { path: 'add', View: Add },
-];
+// select data-time
+const dt = document.querySelector('.date-time');
 
-// color the current link
-const showOnPage = () => {
-  routes.forEach((route) => {
-    if (route.path !== '/') {
-      const current = document.getElementById(route.path);
-      if ((`#${route.path}`) === window.location.hash) {
-        current.classList.add('current');
-      } else {
-        current.classList.remove('current');
-      }
-    }
-  });
-};
-
-// main function to change and update navigation
-const load = async () => {
-  // remove #
-  const fragmentId = window.location.hash.substr(1);
-
-  // get the one whose location is on current hash
-  const route = routes.filter((route) => route.path === fragmentId);
-
-  // give the html class
-  const view = new route[0].View();
-
-  // append the data from the main getDoc
-  document.querySelector('#app').innerHTML = await view.getDoc();
-
-  // change the link color on vchange
-  showOnPage();
-
-  // javascript code for list section
-  if (route[0].path === 'list') {
-    books.createBookFromLocal();
-  }
-
-  // javascript code for add section
-  if (route[0].path === 'add' || route[0].path === '/') {
-    // add button
-    const button = document.getElementById('add-books');
-    // if click add btn, data to local
-    button.addEventListener('click', (e) => {
-      const title = document.getElementById('title').value;
-      const author = document.getElementById('author').value;
-      books.addBook({ e, title, author });
-    });
-  }
-};
+// display every second
+setInterval(() => {
+  dt.innerHTML = formatTs();
+}, 1000);
 
 // if the current is null redirect to add
 if (!window.location.hash) {
   window.location.hash = '#add';
 }
-
 // load for the intital page
-load();
+load(books);
 
 // eventlistener on hash change
 window.addEventListener('hashchange', () => {
-  load();
+  load(books);
 });
 
 // intial color for links
```

Comment:
- [ ] Clicking on the `add new` navigation option suppose to show a form to add a new book, but that's not happening. So please fix that.

![chrome-capture-2022-9-31](https://user-images.githubusercontent.com/68796232/199003956-271312cf-acae-419a-a1cb-40e6e9b60aec.gif)

Reply:
@Omar-Muhamad  No it is actually disabled. No form elements no lists.

Reply:
I tried to add a new task and there is no form to add it so please fix your code so the app can function as expected.

Reply:
Are you running on http server? It is fine here?

Reply:
Are you sure you are running on HTTP server? Here it is fine.
![h](https://user-images.githubusercontent.com/105173981/199007330-4002782a-f61d-4f8a-8335-c4c499b63f30.png)